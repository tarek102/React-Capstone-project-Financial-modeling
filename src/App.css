* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

body {
  background-color: #fb5092;
}

.nav-icons svg {
  color: white;
  font-size: 20px;
  margin-right: 20px;
}

nav h3 {
  color: white;
}

.arrow {
  color: white;
  font-size: 20px;
  margin-left: 15px;
}

.currencies {
  background-color: #c02661;
  margin-right: 15px;
  border-radius: 20px;
  margin-top: 20px;
}

.currencies h2,
.currencies span {
  color: white;
}

.details-card {
  padding: 50px;
  margin-top: 50px;
  background-color: #c02661;
  border-radius: 20px;
}

.title {
  color: white;
  margin-bottom: 20px;
}

.details-card > div {
  display: flex;
  flex-direction: column;
  color: white;
}

.details {
  font-size: 25px;
  margin-bottom: 20px;
}
