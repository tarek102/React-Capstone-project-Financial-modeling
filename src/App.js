import React from 'react';
import { Route, Routes } from 'react-router-dom';
import './App.css';
import Details from './components/Details';
import Header from './components/Header';
import Home from './components/Home';

function App() {
  return (
    <div className="App container">
        <Header />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="details" element={<Details />} />
        </Routes>
    </div>
  );
}

export default App;
