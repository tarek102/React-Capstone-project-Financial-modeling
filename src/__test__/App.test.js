import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../redux/ConfigureStore';
import App from '../App';

describe('Test if App matches snapshot', () => {
  test('matches', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
