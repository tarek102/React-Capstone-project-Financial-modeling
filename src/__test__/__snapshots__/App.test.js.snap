// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if App matches snapshot matches 1`] = `
<div
  className="App container"
>
  <div
    className="navigation"
  >
    <nav
      className="d-flex justify-content-between p-4"
    >
      <div
        className="logo"
      >
        <a
          aria-current="page"
          className="align-self-center arrow active"
          href="/"
          onClick={[Function]}
        >
          <svg
            className="details"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.2426 6.34317L14.8284 4.92896L7.75739 12L14.8285 19.0711L16.2427 17.6569L10.5858 12L16.2426 6.34317Z"
              fill="currentColor"
            />
          </svg>
        </a>
        <h3>
          Crypto currencies
        </h3>
      </div>
      <div
        className="nav-icons"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
          />
          <path
            d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
          />
        </svg>
      </div>
    </nav>
  </div>
  <section
    className="wrapper d-flex flex-column align-items-center"
  >
    <div
      className="hero"
    >
      <h2>
        Welcome to Crypto traders
      </h2>
      <div
        className="search-div"
      >
        <input
          onChange={[Function]}
          placeholder="Search coins"
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
          type="button"
        >
          <a
            className=""
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </button>
      </div>
    </div>
    <div
      className="row cryptos d-flex flex-row justify-content-center"
    />
  </section>
</div>
`;
